grpc/src/main/java/io/stargate/grpc/service/QueryHandler.java

```diff
@@ -192,4 +201,30 @@ private Parameters makeParameters(QueryParameters parameters, Optional<ClientInf
 
     return builder.tracingRequested(parameters.getTracing()).build();
   }
+
+  private CompletionStage<Void> waitForSchemaAgreement() {
+    CompletableFuture<Void> agreementFuture = new CompletableFuture<>();
+    waitForSchemaAgreement(Persistence.SCHEMA_AGREEMENT_WAIT_RETRIES, agreementFuture);
+    return agreementFuture;
+  }
+
+  private void waitForSchemaAgreement(
+      int remainingAttempts, CompletableFuture<Void> agreementFuture) {
+    if (persistence.isInSchemaAgreement()) {
+      agreementFuture.complete(null);
+      return;
+    }
+    if (remainingAttempts <= 1) {
+      agreementFuture.completeExceptionally(
+          new IllegalStateException(
```

Comment:
I think this ends being an `UNKNOWN` error from the user's perspective without any information. It could be nicer to use a `StatusException`.  `Status.DEADLINE_EXCEEDED.withDescriptions("Failed to reach schema agreement after...").toException()` ?

Reply:
Good suggestion, will do 👍